cypress/integration/providers_spec.js

```diff
@@ -0,0 +1,111 @@
+import { shouldBeRedirectedToLogin } from '../support/assertions';
+
+describe('Dashboard Providers Page', () => {
+  describe('When not logged in', () => {
+    it('should redirect to login page', () => {
+      cy.visit('/#/providers');
+      shouldBeRedirectedToLogin();
+    });
+  });
+
+  describe('When logged in', () => {
+    // test provider name
+    const name = 'TESTPROVIDER';
+    beforeEach(() => {
+      cy.login();
+    });
+
+    it('displays a link to view providers', () => {
+      cy.visit('/');
+
+      cy.get('nav li a').contains('Providers').should('exist').as('providers');
+      cy.get('@providers').should('have.attr', 'href', '#/providers');
+      cy.get('@providers').click();
+
+      cy.url().should('include', 'providers');
+      cy.get('.heading--xlarge').should('have.text', 'Providers');
+
+      cy.get('table tbody tr').its('length').should('be.gt', 1);
+    });
+
+    it('providers page displays a button to add a new provider', () => {
+      cy.visit('/#/providers');
+
+      cy.get('.heading--large').should('have.text', 'Provider Overview');
+      cy.get('a').contains('Add a Provider').should('exist').as('addProvider');
+      cy.get('@addProvider').should('have.attr', 'href', '#/providers/add');
+      cy.get('@addProvider').click();
+
+      cy.get('.heading--xlarge').should('have.text', 'Providers');
+      cy.get('.heading--large').should('have.text', 'Create a provider');
+
+      // fill the form and submit
+      cy.get('form div ul').as('providerinput');
+      cy.get('@providerinput').contains('Provider Name').siblings('input').type(name);
+      cy.get('@providerinput').contains('Concurrent Connnection Limit').siblings('input').clear().type(5);
+      cy.get('@providerinput').contains('label', 'Protocol').siblings().children('select')
+      .select('s3', {force: true}).should('have.value', 's3');
+
+      cy.get('@providerinput').contains('Host').siblings('input').type('test-host');
+
+      cy.get('form div input[value=Submit]').click();
+
+      // displays the new provider
+      cy.get('.heading--xlarge').should('have.text', 'Providers');
+      cy.get('.heading--large').should('have.text', name);
+      cy.url().should('include', `#/providers/provider/${name}`);
+
+      // verify the new provider is added to the providers list
+      cy.get('a').contains('Back to Provider').click();
+      cy.get('.heading--large').should('have.text', 'Provider Overview');
+      cy.get('table tbody tr').its('length').should('be.gt', 1);
```

Comment:
If we wanted this assertion to provide a new provider was created, we would need to test for greater than 2, correct? Since the fixture for providers includes 2 items. But I feel like this assertion may be unnecessary anyway since line 65 really proves that the provider was added to the list